packages/client/src/components/button/button.tsx

```diff
@@ -0,0 +1,13 @@
+import styles from './button.module.css'
+
+export const Button = (props: React.ComponentPropsWithoutRef<'button'>) => {
```

Comment:
Можно выбрать из `ComponentPropsWithoutRef<'button'>` нужные аттрибуты и обьединить с пропами компонента

```suggestion
type ButtonProps = {
  someComponentProp: string
} & Pick<React.ComponentPropsWithoutRef<'button'>, 'className', 'type'>

export const Button = (props: ButtonProps) => {
```

Reply:
> При проверке ПР тест падает, его можно удалить

Если удалить тест, ошибка будет в любом случае, так как в проекте будут отсутствовать тесты  
``No tests found, exiting with code 1`` или нет?

Reply:
> Можно выбрать из `ComponentPropsWithoutRef<'button'>` нужные аттрибуты и обьединить с пропами компонента

Понятно, спасибо за пример, но у меня вопрос: "Зачем"? Чтобы ``IntelliSense`` работало лучше выдавая меньшее количество вариантов?  
Я поначалу тоже полагал, что это удобно и в первой версии так примерно и было (правда без свойства ``type``) Но ты мне подсказал тип ``ComponentPropsWithoutRef<'button'>`` который позволяет обойтись без подробного описания свойств компонента и при этом для них работает ``IntelliSense``. Думаю этот вариант более универсален.

Если очень нужно описать все свойства которые мы планируем использовать в компоненте ``button``, предлагаю кому-нибудь, кто чётко представляет себе весь набор этих свойств описать их полностью самостоятельно и внести потом изменения в главную ветку, ибо лично я не уверен в том, какие именно свойства будут нужны, а какие нет.

Reply:
Тест пытается найти текст, которого нет в `App` и падает, можешь изменить тест, чтобы от проходил на компоненте

Reply:
> Тест пытается найти текст, которого нет в `App` и падает, можешь изменить тест, чтобы он проходил на компоненте

Помогло. А я думал что из-за store ошибка будет в любом случае.